src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonedStates = [];
+  let currentState = { ...state };
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    let nextState;
```

Comment:
you don't need this extra variable `nextState`, you can manipulate `currentState`